src/data/Dhis2Events.ts

```diff
@@ -21,3 +29,19 @@ export async function postEvents(api: D2Api, events: Event[]): Promise<Synchroni
 
     return eventsResult;
 }
+
+function setEnrollmentToEvent(
+    events: Event[],
+    options: { existingTeis: TrackedEntityInstance[]; teis: TrackedEntityInstance[] }
+) {
+    const { existingTeis, teis } = options;
+    return teis.length === 0
+        ? events
+        : events.map(event => {
+              const existingTei = existingTeis.find(tei => tei.id === event.trackedEntity);
+              const tei = teis.find(tei => tei.id === event.trackedEntity);
```

Comment:
A map->find is a O(n^2) operation. That's not a problem if we have very few records, is that case here? otherwise, create teiById and existingTeiById and now the loop goes down to O(n)